### ISSUE TITLE
Rule G004 doesn't work with inline instantiated logger

### ISSUE THREAD

**Issue by Spenhouet** [astral-sh/ruff#11031]
Given the following code:
```python
logger = Logger()
logger.warning(f"{database_name} database not available.")
```

the ruff linter will complain about G004 ("Logging statement uses f-string").

But given an inline instantiation of the logger, there is no such warning.
```python
Logger().warning(f"{database_name} database not available.")
```

The "G" rule (flake8-logging-format) needs to be enabled.
My ruff version is 0.3.1.
I searched for G004 but none of the existing issues cover this. The answer might be similar to https://github.com/astral-sh/ruff/issues/6353#issuecomment-1666263576 but I do think that this is something which could be covered.

**Comment by dhruvmanila** [comment]
Yeah, our heuristics doesn't check for a direct instantiation https://github.com/astral-sh/ruff/blob/6979dadf3dcede7e120780645b14a8d2d8d92c24/crates/ruff_python_semantic/src/analyze/logging.rs#L7-L65

We could support this by updating the heuristic to include `logging.Logger()` calls
